apps/src/aichat/types.ts

```diff
@@ -25,11 +27,19 @@ export interface ChatEvent {
   descriptionKey?: keyof typeof ChatEventDescriptions;
 }
 
+export enum TeacherFeedback {
+  CLEAN_DISAGREE = 'clean_disagree', // Flagged a clean message as inappropriate
+  PROFANITY_AGREE = 'profanity_agree', // Thumbs-upped profane message
+  PROFANITY_DISAGREE = 'profanity_disagree', // Thumbs-downed profane message
+}
+
 export interface ChatMessage extends ChatEvent {
   chatMessageText: string;
   role: Role;
   status: ValueOf<typeof AiInteractionStatus>;
   requestId?: number;
+  // If undefined, the teacher took no action or undid their action.
```

Comment:
I wonder if it would be useful to log if a teacher flagged, and then unflagged a message. Could be `clean_agree`.

Reply:
That is a question I asked too, and I chatted about this with Sam and folks and Sam didn't think it would be info that'd be useful to her. I agree / think it would be hard to know if it was a true agree or an accidental toggle/untoggle.

Reply:
Makes sense! Also noting that we'll probably be logging analytics so we'll be able to keep track of this data via Statsig.